### ISSUE TITLE
[CheckCombCycles][CombFolds] Regression in large logical reduces

### ISSUE THREAD

**Issue by uenoku** [llvm/circt#6587]
It takes 7~10 minutes to compile the following chisel code:
```scala
//> using scala "2.13.11"
//> using repository sonatype-s01:snapshots
//> using lib "org.chipsalliance::chisel::6.0.0-M3+117-2372b1c4-SNAPSHOT"
//> using plugin "org.chipsalliance:::chisel-plugin::6.0.0-M3+117-2372b1c4-SNAPSHOT"

//> using options "-unchecked", "-deprecation", "-language:reflectiveCalls", "-feature", "-Xcheckinit", "-Xfatal-warnings", "-Ywarn-dead-code", "-Ywarn-unused", "-Ymacro-annotations"

import chisel3._
import circt.stage.ChiselStage

class Foo extends Module {
  val a = IO(Input(Vec(65536, Bool())))
  val out = IO(Output(Bool()))
  out := a.reduce(_ | _)
}

object Main extends App {
println(
  ChiselStage.emitCHIRRTL(new Foo()))
// 
//   ChiselStage.emitSystemVerilog(
//     gen = new Foo,
//     firtoolOpts = Array("-disable-all-randomization", "-strip-debug-info", "-verbose-pass-executions")
//   )
}
```

CheckCombCycles and OrOp canonicalizers are slow.

**Comment by seldridge** [comment]
Good find and thanks for the report!

**Comment by uenoku** [comment]
Closed by #6593 and #5647.
